weavertest/init.go

```diff
@@ -28,12 +28,43 @@ import (
 	"github.com/ServiceWeaver/weaver/runtime/logging"
 )
 
+// Mode controls where components live and how their methods are invoked.
+type Mode uint8
+
+const (
+	// Local places all components in the same process and uses local procedure calls
+	// for method invocations.
+	// This is the default value.
+	Local Mode = iota
+
+	// Multi places all components in different process (unless explicitly colocated)
+	// and uses RPCs for method invocations on remote components and local procedure calls
+	// for method invocations on colocated components.
+	Multi
+
+	// RPC places all components in the same process and uses RPCs for method invocations.
```

Comment:
nit: maybe we can add RPC after Local, so it's single process local calls, single process remote calls, then multi process